### PR TITLE
fix: ResponseInterface (1)

### LINKED ISSUE #5351: Bug: Undefined method 'setBody'.
### PHP Version

7.4

### CodeIgniter4 Version

4.1.5

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

cli-server (PHP built-in webserver)

### Database

MariaDB

### What happened?

image

### Steps to Reproduce

using "intelephense"
and this my code

```php
return $this->response
            ->setStatusCode(200)
            ->setContentType($file->getMimeType())
            ->setBody(file_get_contents($img))
            ->send();
```

### Expected Output

The notice should not appear.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Ref #4356
Fixes #5351

- fix `ResponseInterface`
- fix `MessageInterface`
- deprecate
   - `CodeIgniter::$useSafeOutput`
   - `CodeIgniter::useSafeOutput()`

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/CodeIgniter.php [hunk 1]
```diff
@@ -1057,7 +1061,7 @@ public function spoofRequestMethod()
      */
     protected function sendResponse()
```
Comment (MGatner) [inline system/CodeIgniter.php:1062]: Can we deprecate this method since now it is literally just "respond send"?

File: system/HTTP/Response.php [hunk 2]
```diff
@@ -188,6 +188,7 @@ public function __construct($config)
 
     /**
      * Turns "pretend" mode on or off to aid in testing.
+     *
      * Note that this is not a part of the interface so
      * should not be relied on outside of internal testing.
```
Comment (MGatner) [inline system/HTTP/Response.php:193]: Let's start annotating these `@testTag` to indicate the class/method is only for testing (following this convention: https://github.com/DaveLiddament/php-language-extensions#testtag). Once we have Attributes we can decide if we want to elevate these to an actual attribute - but at least we can start accumulating them now.

I could probably craft a custom PHPStan rule to work with the annotation in the meantime as well.

**Review by MGatner — CHANGES_REQUESTED** [review]
Much better separation of concerns! A few suggestions.

File: system/HTTP/Response.php (hunk 2 above)
Comment (kenjis) [reply system/HTTP/Response.php:193]: Added `@testTa`

File: system/CodeIgniter.php (hunk 1 above)
Comment (kenjis) [reply system/CodeIgniter.php:1062]: I think it is better the CodeIgniter class has the method, this level of abstraction.

**Review by MGatner — APPROVED** [review]

**Comment by kenjis** [comment]
Added the docs.

I've found this PR is not enough to fix ResponseInterface.
But it seems a lot of changes will be needed, so I'm going to another PR.

**Review by MGatner — APPROVED** [review]
Docs look good

**Pull request merged**